src/Microsoft.Azure.WebJobs.Host/Scale/ScaleMonitorManager.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.WebJobs.Host.Scale
+{
+    internal class ScaleMonitorManager : IScaleMonitorManager
+    {
+        private readonly List<IScaleMonitor> _monitors = new List<IScaleMonitor>();
+        private object _syncRoot = new object();
+
+        public ScaleMonitorManager()
+        {
+        }
+
+        public ScaleMonitorManager(IEnumerable<IScaleMonitor> monitors)
+        {
+            // add any initial monitors coming from DI
```

Comment:
In addition to IListener based monitors mapped to functions, I've designed all this to support more general monitors, e.g. a host level CPU/Memory monitor. Such a monitor isn't tied to a function listener but could be registered with DI and would be picked up here.